### PR TITLE
Warnings

### PR DESCRIPTION
**Pull request by martijnhoekstra**
Eliminate some warnings (deprecations, procedure syntax, feature warnings)

### CODE REVIEW

**Comment by som-snytt** [comment]
For postfix, compare https://github.com/scala/scala/pull/6559/files

**Review by SethTisue — APPROVED** [review]

**Comment by som-snytt** [comment]
I just think there should only be one postfix fixer, the way there's only supposed to be one vampire slayer, and look what happened when they broke that rule. I was only called because soc is not around.

**Comment by martijnhoekstra** [comment]
I dream of a build with 0 warnings, and building under `-feature -deprecation -umchecked`

Maybe for a significant part of remaining warnings scalafix can be employed.

**Comment by hrhino** [comment]
"Dogfooding" scalafix on scalac seems like a really good idea.

File: src/repl-frontend/scala/tools/nsc/interpreter/shell/Pasted.scala [hunk 1]
```diff
@@ -125,7 +125,7 @@ abstract class Pasted(prompt: String, continuePrompt: String, continueText: Stri
    */
   def apply(lines: IterableOnce[String]): Option[String] = {
     isRunning = true
-    try new PasteAnalyzer(lines.toList).run()
+    try new PasteAnalyzer(List.from(lines)).run()
```
Comment (lrytz) [inline src/repl-frontend/scala/tools/nsc/interpreter/shell/Pasted.scala:128]: could be `.to(List)`, which reads more left-to-right

File: src/compiler/scala/tools/nsc/backend/jvm/opt/BoxUnbox.scala [hunk 2]
```diff
@@ -229,9 +229,9 @@ abstract class BoxUnbox {
           val localSlots = Vector.from[(Int, Type)](boxKind.boxedTypes.iterator.map(tp => (getLocal(tp.getSize), tp)))
 
           // store boxed value(s) into localSlots
-          val storeOps = localSlots.toList reverseMap { case (slot, tp) =>
+          val storeOps = localSlots.reverseIterator map { case (slot, tp) =>
             new VarInsnNode(tp.getOpcode(ISTORE), slot)
-          }
+          } to(List)
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/backend/jvm/opt/BoxUnbox.scala:234]: That almost looks like postfix :-)

**Review by lrytz — APPROVED** [review]
Thanks, @martijnhoekstra!

**Comment by SethTisue** [comment]
thank you, Martijn!

further PRs (from anyone reading this) cutting back on warnings even more would be very welcome

**Comment by som-snytt** [comment]
For future (or current) reference, only the varargs `Buffer#append` was deprecated. Appending one element is an alias for `addOne`.

Noticed when diffing stuff not forward-ported to dotty.

I don't know if that implies a previous (or current) limitation of scalafix.

**Pull request merged**